### DISCUSSION TITLE
PHP 7.4

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3089] [General]
Since we're about to release a big update with 19.5 and 20.1, wouldn't it be a good moment to bump to PHP7.4 as minimum requirement?

**Comment by tmotyl** [comment]
Im ok with that. Alternatively we can bump requirements in the next release (with fewer changes)

**Comment by justinbeaty** [comment]
I think we already require 7.4 due to this: https://github.com/OpenMage/magento-lts/pull/2982#issuecomment-1398505657

I would vote to bump it in composer.json.

**Reply by elidrissidev** [reply]
~My local installation only checks for 7.3 in `vendor/composer/platform_check.php`~ nvm that's because of `config.platform.php` in `composer.json` is overriding it.

**Comment by elidrissidev** [comment]
I'm fine with it. Making 7.4 the minimum does mean we can start using [some cool features](https://www.php.net/releases/7_4_0.php).

**Comment by MelissiaKorineckM0u** [comment]
Since we're about to release a big update with 19.5 and 20.1, wouldn't it be a good moment to bump to PHP7.4 as minimum requirement?

**Comment by addison74** [comment]
Debian 10 which is delivered with PHP 7.3 has LTS until June 2024. Unlike Debian 9 where only by compiling the source code you can use a PHP 7.4, on Debian 10 you can install packages of 7.4 and 8.x from Sury. From my point of view, I think it is OK to stay with PHP 7.4 for a while, especially since Debian 11 comes with it and has LTS until June 2026, 3 years from now.

**Reply by fballiano** [reply]
I agree, I wouldn't make php8 mandatory for a while, although php7 it's not officially supported anymore. but also, if a store is on php 7.1 or 7.2, they can update to 7.4 without too much hussle (as far as I've seen) so it's should create too many problems to people.

**Reply by addison74** [reply]
In Magento 2.4.6 the support for PHP 7.4 was dropped.

**Reply by luigifab** [reply]
Some weeks ago I have tested OpenMage 20 with PHP 7.2, still working :)
Okay, I don't have tested all pages and functionalities.

**Comment by WilfordWetzelj** [comment]
Hello fballiano,

It's great to see that you're considering updating to PHP 7.4 as a minimum requirement for the upcoming release. This version of PHP brings many performance improvements, new features, and security enhancements.

However, before making such a change, it's important to consider the potential impact on users who may not have upgraded their PHP versions yet. It's recommended to perform thorough testing to ensure that the application is fully compatible with PHP 7.4 and that there are no breaking changes.

**Reply by elidrissidev** [reply]
For now, apart from the change in `composer.json`, there are no 7.4 specific features used in the codebase so it should still work with 7.3 if you bypass `vendor/composer/platform_check.php`. Yes, thorough testing will need to be performed by users to ensure their extensions are compatible, but from my personal experience when we upgraded to PHP 7.4 2 years ago, it was quite easy as the majority of extensions supported it.

**Reply by tmotyl** [reply]
If some code is not v7.4 compat, there is rector to help https://getrector.com/
